bitwarden_access_manager.py

```diff
@@ -0,0 +1,737 @@
+
+            group_accesses_data = tuple(
+                sorted(self.get_collection_groups(collection_data["groups"]))
+            )
+
+            if len(group_accesses_data) > 0:
+                group_accesses = group_accesses_data
+
+            if collection_id in collections_members:
+                member_accesses = tuple(sorted(collections_members[collection_id]))
+
+            yield Collection(
+                id=collection["id"],
+                external_id=collection["externalId"],
+                member_access=member_accesses,
+                group_access=group_accesses,
+            )
+
+    def get_group_members(self, id: str, name: str) -> Iterable[GroupMember]:
+        members = json.load(self._http_get(f"/public/groups/{id}/member-ids"))
+
+        for member in members:
+            member = json.load(self._http_get(f"/public/members/{member}"))
+            yield GroupMember(
+                member_id=member["id"],
+                member_name=member["name"],
+                group_name=name,
+            )
+
+    def set_member_type(self, type_id: int) -> MemberType:
+        int_to_member_type: Dict[int, MemberType] = {
+            0: MemberType.OWNER,
+            1: MemberType.ADMIN,
+            2: MemberType.USER,
+            3: MemberType.MANAGER,
+            4: MemberType.CUSTOM,
+        }
+        return MemberType(int_to_member_type[type_id])
+
+    def get_members(
+        self,
+    ) -> tuple[List[Member], Dict[str, List[MemberCollectionAccess]]]:
+        data = self._http_get(f"/public/members")
+        members = json.load(data)
+
+        members_result: List[Member] = []
+        collection_access: Dict[str, List[MemberCollectionAccess]] = {}
+
+        for member in members["data"]:
+            type = self.set_member_type(member["type"])
+            m = Member(
+                id=member["id"],
+                name=member["name"],
+                email=member["email"],
+                type=type,
+                access_all=member["accessAll"],
+            )
+            members_result.append(m)
+
+            collections = json.load(self._http_get(f"/public/members/{member['id']}"))[
+                "collections"
+            ]
+            if type != MemberType.OWNER and type != MemberType.ADMIN:
+                for collection in collections:
+                    access = self.check_access(collection["readOnly"])
+
+                    if collection["id"] not in collection_access:
+                        collection_access[collection["id"]] = [
+                            MemberCollectionAccess(name=member["name"], access=access)
+                        ]
+                    else:
+                        collection_access[collection["id"]].append(
+                            MemberCollectionAccess(name=member["name"], access=access)
+                        )
+
+        return members_result, collection_access
+
+    def check_access(self, readonly: bool) -> GroupAccess:
```

Comment:
```suggestion
    def map_access(self, *, readonly: bool) -> GroupAccess:
```
This doesn’t check anything, it just converts the bool to `GroupAccess`. We can also add a `*` to force the caller to use keyword args for everything following it, that makes the call site a bit more obvious: we have to write `self.map_access(readonly=collection["readOnly"])` where you can see at a glance that it is passing in the right value. If somebody would write `self.map_access(collection["canWrite"])` it’s not obvious that that is a bug, but with `self.map_access(readonly=collection["canWrite"])` it is.

Reply:
You're right, it's more obvious using this form https://github.com/ChorusOne/github-access-manager/pull/10/commits/22d7d4486ea2baafbd7a820b73e036b8351c2c8f